darkdetect/_mac_detect.py

```diff
@@ -69,6 +80,38 @@ def isDark():
 def isLight():
     return theme() == 'Light'
 
-#def listener(callback: typing.Callable[[str], None]) -> None:
-def listener(callback):
-    raise NotImplementedError()
+
+def _listen_child():
+    """
+    Run by a child process, install an observer and print theme on change
+    """
+    class Observer(NSObject):
+        def callback_(self, _):
+            try:
+                print(theme(), flush=True)
+            except IOError:
+                os._exit(1)
+    observer = Observer.new()  # Keep a reference of the observer to keep it alive
+    NSDistributedNotificationCenter.defaultCenter().addObserver_selector_name_object_(
+        observer,  # Observer must be kept alive by a different reference
+        "callback:",
+        "AppleInterfaceThemeChangedNotification",
+        None
+    )
+    AppHelper.runConsoleEventLoop()
+
+
+def listener(callback: Callable[[str], None]) -> None:
+    if not _can_listen:
+        raise NotImplementedError()
+    pth = f"import sys; sys.path.insert(0, r'''{Path(__file__).parents[1]}''')"
+    sig = "import signal as s; s.signal(s.SIGINT, s.SIG_IGN)"
+    listen = "import darkdetect as dd; dd._mac_detect._listen_child()"
+    with subprocess.Popen(
+        (sys.executable, "-c", f"{pth}; {sig}; {listen}"),
+        stdout=subprocess.PIPE,
+        # stderr=subprocess.DEVNULL,
+        universal_newlines=True,
```

Comment:
I would avoid a `sys.path` manipulation and instead use the `cwd` argument in `subprocess.Popen`. Suggested change:
```python
    cwd = Path(__file__).parent
    listen = "import _mac_detect as md; md._listen_child()"
    with subprocess.Popen(
        (sys.executable, "-c", f"{listen}"),
        stdout=subprocess.PIPE,
        universal_newlines=True,
        cwd=cwd,
```